### DISCUSSION TITLE
Redesigning Poetry's Schema Handling

### DISCUSSION THREAD

**Discussion by abn** [python-poetry/poetry#4780] [Ideas]
`poetry-schema.json`: I noticed that both projects ([poetry](https://github.com/python-poetry/poetry/blob/master/src/poetry/json/schemas/poetry-schema.json) and [poetry-core](https://github.com/python-poetry/poetry-core/blob/master/poetry/core/json/schemas/poetry-schema.json)) carry a schema file. Is that really required? I guess we could have only one file in **poetry-core** and be consumed by **poetry** when/if required.

_Originally posted by @wagnerluis1982 in https://github.com/python-poetry/poetry/issues/4776#issuecomment-970717361_

**Comment by abn** [comment]
The poetry version of the schema is a hangover from before splitting out poetry core and should be removed and only extensions (or things that do not make sense in core) be defined in `poetry`.

**Comment by wagnerluis1982** [comment]
It seems to me that the sole place which the schema is used is at the function `from poetry.json import validate_object`, the link below show the exact point I traced the usage.

https://github.com/python-poetry/poetry/blob/7c53db9680d021bac99cc366a3bbc88ebbffdf0f/src/poetry/json/__init__.py#L18

Thing is, there is NOT a single USAGE to the function `validate_object`, in fact I removed completely from the working dir the folder `src/poetry/json/` and run `pytest`, not a single test has failed. Which indicates to me the `poetry-schema.json` is being uselessly updated in **poetry** on the last months (more than 6 months at least).

**Reply by neersighted** [reply]
Please feel free to submit a PR deleting the file/references -- it indeed is just dead weight.
